zerver/lib/digest.py

```diff
@@ -149,8 +150,9 @@ def _enqueue_emails_for_realm(realm: Realm, cutoff: datetime.datetime) -> None:
         )
 
 
+@functools.lru_cache(maxsize=500)
```

Comment:
Hmm, should we do like 5000 here? I think that would be safely above the maximum number of streams in a realm, and I think has little downside?

Reply:
I wonder if we also need to set a maximum age in this cache, so that fetches from last week's run cannot be incorrectly used. I guess the different `cutoff_date` handles that, so maybe just worth adding a comment explaining that reasoning.

Reply:
The downside to setting it too big is ballooning memory over time, even for small deployments.  We can try something a little "smarter" in the queue worker, in terms of clearing this cache when we see a new `realm_id` or `cutoff_date`, which would let us set this bigger without impacting the memory footprint of the worker for smaller deploys.

Reply:
Added this logic.